### PR TITLE
Reorganize `Cardano.Ledger.Shelley` and `ShelleyMA` era modules

### PR DESCRIPTION
**Pull request by dnadales**
- Move `ShelleyEra` and associated definitions to `Cardano.Ledger.Shelley`.

- Define `Core` instances in the `ShelleMA` era module.

### CODE REVIEW

**Review by nc6 — APPROVED** [review]
Looks great, thanks Damian!

I'm a little uncertain about pulling e.g. `ValidateScript` to the top, but this is my own fault for making these things era-based, rather than using the specific Script type. So this is perhaps something to address in the future, but I think that this PR takes the best approach to these things as they are defined now.

File: shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs [hunk 1]
```diff
@@ -1,57 +1,52 @@
...
 import Shelley.Spec.Ledger.Coin (Coin)
-import Shelley.Spec.Ledger.Hashing (EraIndependentTxBody, HashAnnotated (..))
-
---------------------------------------------------------------------------------
--- Shelley Era
---------------------------------------------------------------------------------
+import Shelley.Spec.Ledger.Keys (hashWithSerialiser)
+import Shelley.Spec.Ledger.MetaData (MetaData (MetaData), MetaDataHash (MetaDataHash), ValidateMetadata (hashMetadata, validateMetadata), validMetaDatum)
+import Shelley.Spec.Ledger.Scripts (MultiSig)
+import Shelley.Spec.Ledger.Tx (TxBody, ValidateScript (hashScript, validateScript), hashMultiSigScript, validateNativeMultiSigScript)
```
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs]: There are a few lines that I would wrap.

File: shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs [hunk 2]
```diff
@@ -1,57 +1,52 @@
...
+instance
+  (CryptoClass.Crypto c, TxBodyConstraints (ShelleyEra c)) =>
+  ValidateScript (ShelleyEra c)
+  where
+  validateScript = validateNativeMultiSigScript
+  hashScript = hashMultiSigScript
+
+instance CryptoClass.Crypto c => ValidateMetadata (ShelleyEra c) where
+  hashMetadata = MetaDataHash . hashWithSerialiser toCBOR
+  validateMetadata (MetaData m) = all validMetaDatum m
```
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs:62]: Probably unrelated to this PR, but I saw a `deepseq` here in some other implementation. Is that intentional?

File: shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley/Constraints.hs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE EmptyDataDecls #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Ledger.Shelley.Constraints where
+
+import Cardano.Ledger.Compactible (Compactible (..))
+import Cardano.Ledger.Core (AnnotatedData, ChainData, Metadata, Script, SerialisableData, TxBody, Value)
```
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley/Constraints.hs]: Rather long :slightly_smiling_face:

**Review by mrBliss — APPROVED** [review]
LGTM

File: shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs (hunk 2 above)
Comment (nc6) [reply shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs:62]: That was in the Shelley-MA package, related to the new optional scripts. The idea there is that we want to verify that scripts included in the metadata are valid. Scripts are valid by construction, so this check comes down to making sure they're fully deserialised.

**Pull request merged**